Challenge-1/src/agent.ts

```diff
@@ -0,0 +1,109 @@
+// Importing necessary modules and constants
+import {
+  BlockEvent,
+  Finding,
+  Initialize,
+  HandleBlock,
+  HealthCheck,
+  HandleTransaction,
+  HandleAlert,
+  AlertEvent,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+} from "forta-agent";
+
+import { ethers } from "forta-agent";
+
+import {
+  CREATE_BOT_FUNCTION,
+  UPDATE_BOT_FUNCTION,
+  BOT_DEPLOYED_ADDRESS,
+  BOT_UPDATE_EVENT,
+  NETHERMIND_DEPLOYER_ADDRESS,
+} from "./constants";
+
+// Function to handle transactions related to bot creation and update
+export function provideTransaction(
+  botCreation: string,
+  botUpdate: string,
+  botDeployedAddress: string,
+  botUpdateEvent: string,
+  nethermindDeployerAddress: string,
+): HandleTransaction {
+  return async function handleTransaction(tx: TransactionEvent) {
+    const finding: Finding[] = [];
+
+    // Filter transactions related to bot creation and update
+    const botCreationAlert = tx.filterFunction(botCreation, botDeployedAddress);
+
+    const botUpdateAlert = tx.filterFunction(botUpdate, botDeployedAddress);
```

Comment:
Just suggesting a more descriptive name for the variables
```suggestion
    const botCreationFunctionCalls = tx.filterFunction(botCreation, botDeployedAddress);

    const botUpdateFunctionCalls = tx.filterFunction(botUpdate, botDeployedAddress);
```

Reply:
This is just a suggestion, but note that the `.filterFunction` can also take an array of function signatures, meaning you could have this

```suggestion
    const functionCalls = tx.filterFunction([botCreation, botUpdate], botDeployedAddress);
```